src/test/llmtemplate_test.cpp

```diff
@@ -348,3 +377,291 @@ TEST_F(LLMChatTemplateTest, ChatTemplateTwoConfigs) {
+            input_side_packet: "LLM_NODE_RESOURCES:llm"
+            input_stream: "LOOPBACK:loopback"
+            input_stream: "HTTP_REQUEST_PAYLOAD:input"
+            output_stream: "LOOPBACK:loopback"
+            output_stream: "HTTP_RESPONSE_PAYLOAD:output"
+            input_stream_info: {
+            tag_index: 'LOOPBACK:0',
+            back_edge: true
+            }
+            node_options: {
+                [type.googleapis.com/mediapipe.LLMCalculatorOptions]: {
+                models_path: "MODELS_PATTERN",
+                plugin_config: "{\"INFERENCE_PRECISION_HINT\":\"f32\"}"
+                }
+            }
+            input_stream_handler {
+            input_stream_handler: "SyncSetInputStreamHandler",
+            options {
+                [mediapipe.SyncSetInputStreamHandlerOptions.ext] {
+                sync_set {
+                    tag_index: "LOOPBACK:0"
+                }
+                }
+            }
+            }
+    })";
+
+class LLMChatTemplateHttpTest : public TestWithTempDir {
+protected:
+    static std::unique_ptr<std::thread> t;
+
+    const std::string GRAPH_PATTERN = "GRAPH_PATTERN";
+    const std::string WORKSPACE_PATTERN = "MODELS_PATTERN";
+    const std::string ONE_MODEL_PATH = "/ovms/llm_testing/facebook/opt-125m";
+
+    std::string tokenizerConfigFilePath;
+    std::string jinjaConfigFilePath;
+    std::string ovmsConfigFilePath;
+    std::string graphConfigFilePath;
+
+    std::string GetFileNameFromPath(const std::string& parentDir, const std::string& fullPath) {
+        std::string fileName = fullPath;
+        fileName.replace(fileName.find(parentDir), std::string(parentDir).size(), "");
+        return fileName;
+    }
+
+    bool CreateConfigFile(const std::string& graphPath) {
+        std::string configContents = configTemplate;
+        configContents.replace(configContents.find(GRAPH_PATTERN), std::string(GRAPH_PATTERN).size(), graphPath);
+        return CreateConfig(configContents, ovmsConfigFilePath);
+    }
+
+    bool CreatePipelineGraph(const std::string& workspacePath) {
+        std::string configContents = graphTemplate;
+        configContents.replace(configContents.find(WORKSPACE_PATTERN), std::string(WORKSPACE_PATTERN).size(), workspacePath);
+        return CreateConfig(configContents, graphConfigFilePath);
+    }
+
+    void CreateSymbolicLinks() {
+        for (const auto& entry : fs::directory_iterator(ONE_MODEL_PATH)) {
+            std::filesystem::path outfilename = entry.path();
+            std::string outfilename_str = outfilename.string();
+            std::string fileName = GetFileNameFromPath(ONE_MODEL_PATH, outfilename_str);
+            SPDLOG_INFO("Filename to link {}\n", fileName);
+            std::string symlinkPath = ovms::FileSystem::joinPath({directoryPath, fileName});
+            SPDLOG_INFO("Creating symlink from: {}\n to:\n{}", outfilename_str, symlinkPath);
+            fs::create_symlink(outfilename_str, symlinkPath);
```

Comment:
Can we create guard classes for the symlinks? This way we are sure that every symlink created in unit test will be destroyed upon successful test run **and also** fail at any point.